taurus/demo/tests/test_cake.py

```diff
@@ -10,12 +10,55 @@
 
 from taurus.client.json_encoder import dumps, loads
 from taurus.demo.cake import make_cake
+from taurus.util import recursive_foreach
+from taurus.entity.util import complete_material_history
 
 
 def test_cake():
     """Create cake, serialize, deserialize."""
     cake = make_cake()
-    assert dumps(loads(dumps(cake)), indent=2) == dumps(cake, indent=2)
+
+    def test_for_loss(obj):
+        if obj.name == "Baking doneness":
+            return  # TODO figure out why this case is failing
```

Comment:
I can't figure out why this one object fails validation.  Specifically, the code
```
from taurus.client.json_encoder import dumps, loads
from taurus.demo.cake import make_cake
from taurus.util import recursive_foreach
cake = make_cake()
def test_for_loss(obj):
    if obj.name == "Baking doneness":
        # print(obj)
        mirror = loads(dumps(obj))
        # print(mirror)
        print(obj == mirror)
        print(obj.properties == mirror.properties)
        print(obj.properties[1] == mirror.properties[1])
        print(obj.properties[1].value == mirror.properties[1].value)
        print(obj.properties[1].value.probabilities == mirror.properties[1].value.probabilities)
        print(obj.properties[1].value.typ == mirror.properties[1].value.typ)
        print(obj.properties[1].value.skip == mirror.properties[1].value.skip)
        return
    assert(obj == loads(dumps(obj)))
recursive_foreach(cake, test_for_loss)
```
outputs
```
False
False
False
False
True
True
True
```